card/lib/card/location.rb

```diff
@@ -0,0 +1,45 @@
+# -*- encoding : utf-8 -*-
+# helper for urls, links, redirects, and other location related things.
+#  note: i'm sure this isn't the optimal name..
+
+module Card::Location
+
+ # -----------( urls and redirects from application.rb) ----------------
+
+  # TESTME
+  def page_path title, opts={}
+    
+    format = opts[:format] ? ".#{opts.delete(:format)}"  : ''
+    query  = opts.present? ? "?#{opts.to_param}"         : ''
+    card_path "#{title.to_name.url_key}#{format}#{query}"
+  end
+
+  def card_path rel #should be in smartname?
+    rel_path = Card===rel ? rel.cardname.url_key : rel.to_s
+    if rel_path =~ /^\//
+      rel_path
+    else
+      "#{ Cardio.config.relative_url_root }/#{ rel_path }"
+    end
+  end
+
+  def card_url rel #should be in smartname?
+    if rel =~ /^https?\:/
+      rel
+    else
+      "#{ Card::Env[:protocol] }#{ Card::Env[:host] }#{ card_path rel }"
+    end
+  end
+
+
+  # Links ----------------------------------------------------------------------
+
+  def link_to_page( text, title=nil, options={})
+    title ||= text
+    url_options = {}
+    [:type, :view].each { |k| url_options[k] = options.delete(k) if options[k] }
+    url = page_path( title, url_options )
+    link_to text, url, options
+  end
+
```

Comment:
I would love to have location wrapped into the mod system.

It used to be included in controllers and mailers if not elsewhere.  I think you managed to remove it from mailers without problem.  And you separated out the controller stuff a bit, too, right?  (Can't see all that here). If so, I'd guess there are some calls like wagn_url that would need to be tweaked.

Right, I'd definitely like to be thinking MVP (minimal viable product) here so that we can get this stuff merged in.  I just don't know whether the location stuff is currently fully functional.